### PR TITLE
changes to storybook and vercel yaml files, error in storybook.yml

### LINKED ISSUE #6772: [OTHER] Add Repository Condition to GitHub Actions to Prevent Execution in Forked Repositories
### What would you like to share?

Upon syncing the forked repository, I observed that the `Storybook` and `Deploy releases to Vercel` GitHub Actions were being triggered and subsequently failing. Here's a screenshot for reference:
image

I discovered that these two Actions lack a conditional check to ensure they only run in the `EddieHubCommunity/LinkFree` repository, which leads to these failures in the forked repository. 

To remedy this, I suggest we add the following condition to both GitHub Actions. This will prevent them from executing when the repository is a fork, thereby avoiding unnecessary failures:

```yaml
# .github/workflows/storybook.yml
jobs:
  build:
    if: github.repository == 'EddieHubCommunity/LinkFree'
    runs-on: ubuntu-latest
    steps:
        # steps here...
 
  deploy:
    if: github.repository == 'EddieHubCommunity/LinkFree' && github.event_name != 'pull_request'
    # other properties here...
    steps:
      # steps here...
```

```yaml
# .github/workflows/vercel.yml
jobs:
  deploy:
    if: github.repository == 'EddieHubCommunity/LinkFree'
    runs-on: ubuntu-latest
    steps:
        # steps here...
```

Upon triaging of this issue, I would be glad to implement these changes. Thanks!

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by poorvijn**
## Fixes Issue

fixes #6772

## Check List (Check all the applicable boxes) 

- [ ] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] This PR does not contain plagiarized content.
- [ ] The title of my pull request is a short description of the requested changes.

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/6876"></a>

### CODE REVIEW

File: .github/workflows/storybook.yml [hunk 1]
```diff
@@ -53,10 +54,11 @@ jobs:
       name: github-pages
       url: ${{ steps.deployment.outputs.page_url }}
     runs-on: ubuntu-latest
-    if: github.event_name != 'pull_request'
+    if: github.repository == 'EddieHubCommunity/LinkFree' && github.event_name != 'pull_request'
     needs: build
```
Comment (weilincheng) [inline .github/workflows/storybook.yml:58]: Hello @poorvijn, I initially suggested adding the conditional check in #6772 without realizing that the `deploy` steps are dependent on the `build` steps due to the `needs: build` condition. Therefore, the additional conditional check `github.repository == 'EddieHubCommunity/LinkFree'` might not be necessary here. If the `build` steps are skipped, the `deploy` steps should automatically be skipped as well. These are just my thoughts. Feel free to give it a try. 😄

Comment (poorvijn) [reply .github/workflows/storybook.yml:58]: surely going to check it out !

**Comment by eddiejaoude** [comment]
Is this ready for review yet? If so we can take it out of draft

Are there any more Actions that need this check? I think they should all not run on the fork

**Comment by poorvijn** [comment]
I think it's good to go.

**Review by eddiejaoude — APPROVED** [review]
Looks good 👍

**Pull request merged**